lib/string_reverse.rb

```diff
@@ -1,6 +1,21 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: Linear, or O(n), where n is the number of characters inside the string
+# Space complexity: Constant or O(1), because additional space doesn't
+# depend on array size or input
 def string_reverse(my_string)
-  raise NotImplementedError
+  return my_string if my_string == nil
+
+  i = 0
+  j = my_string.length - 1
+
+  while i < j
+    b = my_string[i]
+    my_string[i] = my_string[j]
+    my_string[j] = b
+
+    i += 1
+    j -= 1
+  end
+
+  return my_string
 end

```

Comment:
Nice work on this method! Very clear and easy to follow. Did you write tests for this code?